pkgs/tools/security/vault/vault-bin.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, unzip }:
+
+stdenv.mkDerivation rec {
+  name = "vault-${version}";
+  version = "1.1.3";
+
+  src = fetchurl {
+    url = "https://releases.hashicorp.com/vault/${version}/vault_${version}_linux_amd64.zip";
+    sha256 = "293b88f4d31f6bcdcc8b508eccb7b856a0423270adebfa0f52f04144c5a22ae0";
+  };
+
+  buildInputs = [ unzip ];
+
+  buildCommand = ''
```

Comment:
It's better to define installPhase instead. Unzip should be done automatically.

Reply:
I tried using `InstallPhase` alone but  nix-build complain abount not knowing how to unpack the archive. Adding `unpackPhase` fixes it. replacing `buildCommand` only add more steps to the build so i thing keeping it is better .

Reply:
I think if you add unzip to nativeBuildInputs it should work, take a look here for an example: https://github.com/NixOS/nixpkgs/blob/c126510809c158477690118048bbde7cd83c174b/pkgs/development/interpreters/dart/default.nix#L52

Reply:
works, Thanks